### PR TITLE
Bugfix/Cleanup: Piranha was escaping and unescaping new line characters when not required 

### PR DESCRIPTION
**Pull request by ketkarameya**
Sometimes we need to escape new line characters when it is part of the tree-sitter query. i.e. convert `\n` to `\\n`. 
Problem is that, I was using the method `substitute_tags` for both substituting tags in a replacement pattern and in a tree-sitter query. There was a bug because of this. 
To fix this bug, I basically added a boolean flag indicating wether the input is a tree-sitter query . This flag determines if a  new line character is escaped in the substituted value.

### CODE REVIEW

**Comment by ketkarameya** [comment]
I dont see a test for this.  This was just some confusing and convoluted logic that I straightened.

File: polyglot/piranha/test-resources/java/new_line_character_used_in_string_literal/expected/SomeClass.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.uber.piranha;
+
+class SomeClass {
+  
+  void someMethod(String s){      
+      assert("Hello \n World".equals(s));
```
Comment (lazaroclapp) [inline polyglot/piranha/test-resources/java/new_line_character_used_in_string_literal/expected/SomeClass.java:19]: Presumably, without the change, this becomes:

```
  void someMethod(String s){      
      assert("Hello \\n World".equals(s));
```

correct?

Or is it:

```
  void someMethod(String s){      
      assert("Hello 
 World".equals(s));
```

?

Comment (ketkarameya) [reply polyglot/piranha/test-resources/java/new_line_character_used_in_string_literal/expected/SomeClass.java:19]: yes. It's the later.

File: polyglot/piranha/src/utilities/tree_sitter_utilities.rs [hunk 2]
```diff
@@ -314,6 +314,11 @@ fn _get_tree_sitter_edit(
   }
 }
 
+/// As input, `substitute_tags` requires the (i) `input_string`, (ii) substitutions for the tags (i.e. @<var>) 
+/// and whether the input string is a tree-sitter query (else it is considered as a replacement pattern).
+/// `substitute_tags` replaces the all the occurrences of a specific tag  with the corresponding string values 
+/// specified in `substitutions`, 
+/// Not it escapes newline characters for tree-sitter-queries.
```
Comment (lazaroclapp) [inline polyglot/piranha/src/utilities/tree_sitter_utilities.rs:321]: `Not` -> `Note`, I assume. Also, isn't there a standard way to document arguments in Rust doc comments? https://doc.rust-lang.org/rust-by-example/meta/doc.html (the example there seems to suggest there is)

Comment (ketkarameya) [reply polyglot/piranha/src/utilities/tree_sitter_utilities.rs]: Updated the comment to follow the documentation guidelines.

**Review by lazaroclapp — APPROVED** [review]
LGTM!

**Pull request merged**